packages/CodeGenLib/src/Database/manage-metadata.ts

```diff
@@ -1561,7 +1663,22 @@ export class ManageMetadataBase {
           // For other cases, just add 's'
           return singularName + 's';
       }
-   } 
-   
+   }
+
+   /**
+    * Executes the given SQL query using the given DataSource object.
+    * If the appendToLogFile parameter is true, the query will also be appended to the log file. 
+    * @param ds - The DataSource object to use to execute the query. 
+    * @param query - The SQL query to execute.
+    * @param appendToLogFile - If true, the query will be appended to the log file. 
+    * @param description - A description of the query to append to the log file.
+    * @returns - The result of the query execution.
+    */
+   private async executeSQLQuery(ds: DataSource, query: string, appendToLogFile?: boolean, description?: string): Promise<any> {
```

Comment:
This function name is super similar to one in the sql .js file so I'd say call it something like `executeAndLogMetadataSQL` please. Otherwise looks good

Reply:
@JS-BC  per above latest comments, very close now. Assuming you've tested this and it works :), after you fix the above minor things, go ahead and merge